components/board/server.go

```diff
@@ -207,6 +207,34 @@ func (s *serviceServer) GetDigitalInterruptValue(
 	return &pb.GetDigitalInterruptValueResponse{Value: val}, nil
 }
 
+func (s *serviceServer) StreamTicks(
+	req *pb.StreamTicksRequest,
+	server pb.BoardService_StreamTicksServer,
+) error {
+	b, err := s.coll.Resource(req.Name)
+	if err != nil {
+		return err
+	}
+
+	ticksChan := make(chan Tick, 1024)
+
+	err = b.StreamTicks(server.Context(), req.PinNames, ticksChan, nil)
+	if err != nil {
+		return err
+	}
+	for {
+		select {
+		case <-server.Context().Done():
+			return server.Context().Err()
+		case msg := <-ticksChan:
```

Comment:
Suppose the context is done but `ticksChan` is so full that there's always something in it. I think we might stream ticks forever even though we're supposed to shut down. At the top of the for loop, I'd check just `server.Context().Done()` by itself, with the default of coming down here and doing the rest of the loop as you've written it.